lib/rules/declaration-property-value-no-unknown/index.js

```diff
@@ -0,0 +1,111 @@
+'use strict';
+
+const { isPlainObject } = require('is-plain-object');
+const { fork, parse } = require('css-tree');
+
+const declarationValueIndex = require('../../utils/declarationValueIndex');
+const optionsMatches = require('../../utils/optionsMatches');
+const report = require('../../utils/report');
+const ruleMessages = require('../../utils/ruleMessages');
+const validateOptions = require('../../utils/validateOptions');
+const isCustomProperty = require('../../utils/isCustomProperty');
+const isStandardSyntaxValue = require('../../utils/isStandardSyntaxValue');
+const isStandardSyntaxProperty = require('../../utils/isStandardSyntaxProperty');
+const isStandardSyntaxDeclaration = require('../../utils/isStandardSyntaxDeclaration');
+const { isRegExp, isString } = require('../../utils/validateTypes');
+
+const ruleName = 'declaration-property-value-no-unknown';
+
+const messages = ruleMessages(ruleName, {
+	rejected: (property, value) => `Unexpected unknown value "${value}" for property "${property}"`,
+});
+
+const meta = {
+	url: 'https://stylelint.io/user-guide/rules/declaration-property-value-no-unknown',
+};
+
+/** @type {import('stylelint').Rule} */
+const rule = (primary, secondaryOptions) => {
+	return (root, result) => {
+		const validOptions = validateOptions(
+			result,
+			ruleName,
+			{ actual: primary },
+			{
+				actual: secondaryOptions,
+				possible: {
+					ignoreProperties: [isString, isRegExp],
+					ignoreValues: [isString, isRegExp],
```

Comment:
Can I ignore a regex for a specific property?

e.g. for the property `filter` ignore anything that starts with `progid`

Reply:
Good catch. It seems good the `ignoreProperties` option supports an object instead of an array, for example:

```json5
{
  "ignoreProperties": {
    "filter": "/^progid/",
    "top": "/.*/"  // any values match
  }
}
```

Any thoughts?

Reply:
As long as the name is consistent with the other rules's options. it's OK with me.

I suppose `ignoreValues` is there for global values.
The only example that comes to mind is `-moz-initial`.
If we supported `"ignoreProperties": { "*": "-foo-revert" }` could we drop `ignoreValues`?

Reply:
`ignoreProperties` with an object is [also supported by `unit-allowed-list`](https://stylelint.io/user-guide/rules/unit-allowed-list/#ignoreproperties--unit-property-regex-regexpropertyregexregex-), so I think it is consistent with other rules.

As you say, we could unify `ignoreValues` to `ignoreProperties` with an object. For example:

```json5
{
  "ignoreProperties": {
    "/.*/": "-foo-revert" // ignore the value of any properties
  }
}
```

Reply:
@Mouvedia This is a great suggestion. 

Let's unify `ignoreValues` to `ignoreProperties` with an object as @ybiquitous said.

Reply:
Done! d5a5ca88